terraform/Makefile

```diff
@@ -1,5 +1,22 @@
+ifeq ($(origin .RECIPEPREFIX), undefined)
+  $(error This Make does not support .RECIPEPREFIX. Please use GNU Make 4.0 or later)
```

Comment:
Why?  This seems unnecessary and impedes portability.  I wouldn't be able to run this locally as I have `GNU Make 3.81`.

Reply:
The argument was found in this [post](https://tech.davis-hansson.com/p/make/) as I was doing some `make` research. The gist is that the recipe prefix can avoid any errors brought by tabs or spaces.